src/lambdas/breeds-get.ts

```diff
@@ -0,0 +1,63 @@
+import fetch from 'node-fetch'
+import * as constants from './shared/constants'
+import { Response } from './shared/types'
+
+interface BreedResponse extends Response {
+  body: string[]
+}
+
+interface DogBreeds {
+  // message type is any because this can either be an object (successful response)
+  // or a string (failure)
+  message: any
+  status: string
+  code: number
+}
+
+// Method to flatten response from Breed API
+// input should be a list of key-value pairs
+// where key = breed-type in string
+// and value = string array of sub-breed types (0 or more)
+const flatten = (data: any): string[] => {
+  const parsedData: string[] = []
+
+  Object.keys(data).forEach(key => {
+    const subBreeds = data[key]
+    if (subBreeds.length > 0) {
+      subBreeds.forEach((subBreed: string) => {
+        parsedData.push(`${subBreed} ${key}`)
+      })
+    } else {
+      parsedData.push(key)
+    }
+  })
+
+  return parsedData
+}
+
+export async function handler(): Promise<BreedResponse> {
+  try {
+    const res = await fetch(constants.DOG_BREEDS_API)
+
+    const payload: DogBreeds = await res.json()
+    // handle parsing errors
+    if (payload.status === 'error') {
```

Comment:
If there were an error on `fetch`, the `catch` block would get triggered, so this code will never run.

Reply:
This is to handle cases where the API responded with a valid response but the response didn't contain the breeds. For example if path is something like 'https://dog.ceo/api/breeds/list/incorrectpath', it's a 404 and the API responds with {"status":"error","message":"No route found for \"GET \/api\/breeds\/list\/incorrectpath\" with code: 0","code":404}. In such cases, the fetch was successful but the response didn't contain what we needed. Therefore, control should quit and present the user with the error immediately. If this code block didn't exist, then exception will occur when this payload is passed on to the flatten function.

Reply:
> If there were an error on `fetch`, the `catch` block would get triggered, so this code will never run.

This is to handle cases where the API responded with a valid response but the response didn't contain the breeds. For example if path is something like 'https://dog.ceo/api/breeds/list/incorrectpath', it's a 404 and the API responds with {"status":"error","message":"No route found for "GET /api/breeds/list/incorrectpath" with code: 0","code":404}. In such cases, the fetch was successful but the response didn't contain what we needed. Therefore, control should quit and present the user with the error immediately. If this code block didn't exist, then exception will occur when this payload is passed on to the flatten function.

Reply:
You're right, sorry. We still know `message` will be one of two types, so no need for `any`.

Reply:
message type resolved to message: UnflattenedDogBreed | string